logic/strategy/src/one_hop_connection.rs

```diff
@@ -0,0 +1,358 @@
+//! ## One hop connection strategy
+//! This strategy is designed to maintain 1-hop pathes to certain destination peers,
+//! where there is at least one relay node that is not part of the destination peer set.
+//!
+//! There should be outgoing channels to the destination peers.
+use async_trait::async_trait;
+use chain_actions::channels::ChannelActions;
+use futures::StreamExt;
+use hopr_db_sql::api::peers::PeerSelector;
+use hopr_db_sql::HoprDbAllOperations;
+use hopr_internal_types::channels::{ChannelDirection, ChannelEntry};
+use hopr_primitive_types::prelude::*;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::collections::HashMap;
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::{Add, Sub};
+use std::str::FromStr;
+use tracing::{debug, error, info};
+use validator::Validate;
```

Comment:
Pls split the workspace specific and non-workspace dependencies into separate groups.